src/coreComponents/linearAlgebra/utilities/ReverseCutHillMcKeeOrdering.hpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ReverseCutHillMcKeeOrdering.hpp
+ */
+
+#ifndef GEOS_LINEARALGEBRA_UTILITIES_REVERSECUTHILLMCKEEORDERING_HPP_
+#define GEOS_LINEARALGEBRA_UTILITIES_REVERSECUTHILLMCKEEORDERING_HPP_
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class ReverseCutHillMcKeeOrdering
```

Comment:
Would you consider making this a namespace with a single entry point (`computePermutation`), and placing all `private` functions as free functions in the `.cpp` file (keep `static` on them to avoid unneeded external linkage)?

Reply:
Done, I placed all the private functions as free functions in the cpp file.